### ISSUE TITLE
New Rule: unexported-interface (flag exported interfaces with no exported usage)

### ISSUE THREAD

**Issue by Acollie** [mgechev/revive#1717]
## Is your feature request related to a problem? Please describe.

When refactoring or growing a package, it's easy to leave interfaces declared as exported (capital-letter name) even though they are only ever referenced internally. This bloats the public API surface, encourages downstream packages to depend on details that were never intended to be public, and adds noise to godoc.

Example of an interface that is exported but never used outside unexported call-sites:
```go
package store

// Exported, but the only consumers are unexported helpers in this package.
type Persister interface {
    Save(ctx context.Context, value []byte) error
}

func newDefaultStore(p Persister) *store { ... }

type store struct {
    persister Persister
}
```

Nothing in the package's exported API references `Persister`, so external packages cannot meaningfully use it. It should be `persister`.

This is hard to spot in review once a package grows, and tools like `unparam` don't catch it.

## Describe the solution you'd like

Name: `unexported-interface`

For each exported interface declared in an importable file (skip test files and `main`), walk every file in the package and look for references from the exported API surface:

- exported function/method signatures (methods only count when the receiver is exported);
- exported fields of exported structs;
- methods of other exported interfaces;
- exported `var`/`const`/type-alias declarations.

Function bodies are ignored — types used only inside bodies are not part of the API. If no exported declaration references the interface, report it.

```go
// MATCH: exported interface Persister is not used by any exported declaration; consider unexporting it
type Persister interface {
    Save(ctx context.Context, value []byte) error
}
```

No configuration needed for the first pass. Suggested category: `unexported-type-in-api` (same as `unexported-return`). Confidence around `0.8`. Opt-in (not in `defaultRules`) since revive only sees one package at a time and cannot rule out cross-package usage — the rule is a hint, not a proof.

## Describe alternatives you've considered

- `unexported-return` is the closest existing rule but solves the dual problem (an exported function returning an unexported type), not unused-but-exported interfaces.
- `unparam` performs whole-program analysis but is a separate tool and doesn't focus on interface visibility.
- Manual review — unreliable as the package grows.

## Additional context

Happy to open a PR — I already have a working implementation with tests covering exported func params/results, exported struct fields, embedded interfaces, exported type aliases, exported `var`s, methods on exported vs. unexported receivers, and the corresponding negative cases.

**Comment by alexandear** [comment]
In general, it is impossible to reliably identify exported interfaces with no exported usage.

An exported interface can still be intentionally public even when no exported symbol in the same package references it directly, since external packages may still import and use that type.

I do not want to create a rule that fights against Go’s design. For your specific case, it is better to build a custom `go vet` analyzer.

**Comment by Acollie** [comment]
That's fair view 🙇 I totally understand
